utils/chroma_store.py

```diff
@@ -0,0 +1,164 @@
+
+        # The name must not be a valid IP address
+        try:
+            ipaddress.ip_address(collection_name)
+        except ValueError:
+            pass
+        else:
+            err_msg = f"The collection name must not be a valid IP address, got {collection_name}"
+            return ValidateCollectionName(is_valid=False, err_msg=err_msg)
+
+        return ValidateCollectionName(is_valid=True, err_msg="")
+
+    def _get_or_create_collection(
+        self,
+        collection_name: str,
+        embedding_function: Optional[EmbeddingFunction] = None,
+        metadata: Optional[CollectionMetadata] = None,
+    ) -> None:
+        """Function to create or get a collection.
+
+        Args:
+            collection_name (str): Name of collection
+            embedding_function (Optional[EmbeddingFunction], optional): Embedding function to use. Defaults to None.
+            metadata (Optional[CollectionMetadata], optional): Additional metadata for collection. Defaults to None.
+
+        Raises:
+            ValueError: If `collection_name` is invalid
+        """
+        validation = self.validate_collection_name(collection_name)
+        if not validation.is_valid:
+            raise ValueError(validation.err_msg)
+
+        # If you supply an embedding function, you must supply it every time you get the collection.
+        # By default, all-MiniLM-L6-v2 model is as embedding function
+        self._collection = self._client.get_or_create_collection(
+            name=collection_name,
+            embedding_function=embedding_function,
+            metadata=metadata,
+        )
+
+    def list_collection_names(self) -> List[str]:
+        """List all the collection names in chromadb.
+
+        Returns:
+            List[str]: List containing names of collections
+        """
+        collections = self._client.list_collections()
+        if not collections:
+            return []
+        return [collection.name for collection in collections]
+
+    def add_texts(
+        self,
+        collection_name: str,
+        texts: List[str],
+        ids: Optional[List[str]] = None,
+        metadatas: Optional[List[dict]] = None,
+        embedding_function: Optional[EmbeddingFunction] = None,
+    ) -> None:
+        """Add documents to collection.
+
+        Args:
+            collection_name (str): Name of collection to use for adding documents
+            texts (List[str]): List of textd to be added to the collection
+            ids (Optional[List[str]], optional): Optional list of IDs for documents.
+            metadatas (Optional[List[dict]], optional): Optional list of metadatas for documents. Defaults to None.
+            embedding_function (Optional[EmbeddingFunction], optional): Embedding function to be used by collection. Defaults to None.
+        """
+        if self._collection is None:
+            self._get_or_create_collection(collection_name, embedding_function)
+
+        # A unique id for each document
```

Comment:
Is this necessary and should the `ids` parameter be optional? Since the scrape data step has test to ensure that the `uuids` column is present.